manifests/server/config.pp

```diff
@@ -205,48 +205,57 @@
     }
   }
 
-  if $facts['os']['family'] == 'RedHat' {
-    if $facts['os']['release']['full'] =~ /^7|^8/ or $facts['os']['name'] == 'Fedora' {
+  # RHEL 7 and 8 both support drop-in files for systemd units.  The old include directive is deprecated and may be removed in future systemd releases.
+  # Gentoo also supports drop-in files.
+  if $facts['os']['family'] in ['RedHat', 'Gentoo'] and $facts['service_provider'] == 'systemd' {
       # Template uses:
-      # - $::operatingsystem
+      # - $facts['os']['family']
+      # - $facts['os']['name']
+      # - $facts['os']['release']['major']
       # - $service_name
       # - $port
       # - $datadir
-      file { 'systemd-override':
-        ensure  => file,
-        path    => "/etc/systemd/system/${service_name}.service",
-        owner   => root,
-        group   => root,
-        content => template('postgresql/systemd-override.erb'),
-        notify  => [Exec['restart-systemd'], Class['postgresql::server::service']],
-        before  => Class['postgresql::server::reload'],
-      }
-      exec { 'restart-systemd':
-        command     => 'systemctl daemon-reload',
-        refreshonly => true,
-        path        => '/bin:/usr/bin:/usr/local/bin',
-      }
+      # - @extra_systemd_config
+
+    if (versioncmp($facts['puppetversion'], '5.0.0') <= 0) {
```

Comment:
I think you still need to declare the systemd restart. It's also on all versions older than 6.1.0
```suggestion
    if (versioncmp($facts['puppetversion'], '6.1.0') < 0) {
      exec { 'restart-systemd':
        command     => 'systemctl daemon-reload',
        refreshonly => true,
        path        => '/bin:/usr/bin:/usr/local/bin',
      }
```